src/components/follow-them/follow-them.js

```diff
@@ -2,70 +2,23 @@ import React, { useEffect } from 'react';
 import 'styles.css';
 import avatar from 'assets/defaultImg.png';
 import './follow-them.css';
-import { useAuth } from 'context/authContext';
 import { doc, updateDoc, arrayUnion, arrayRemove } from '@firebase/firestore';
-import { db } from 'firebase';
+import { db } from '../../firebase';
+import { useSelector, useDispatch } from 'react-redux';
+import { addToFollowing, addToFollowers, removeFromFollowers, removeFromFollowing } from 'redux/slices/userFunctions';
 
 export default function FollowThem({ userObj }) {
     const { fullName, username, following, uid, id, followers, profilepic } = userObj;
-    const { user, setUser } = useAuth();
-
-    const addToFollowing = async () => {
-        try {
-            const userRef = doc(db, `users/${user.id}`);
-            const res = await updateDoc(userRef, {
-                following: arrayUnion(uid)
-            })
-            setUser({ ...user, following: [...user.following, uid] });
-            localStorage.setItem("user", JSON.stringify({ ...user, following: [...user.following, uid] }));
-        } catch (e) {
-            console.log(e)
-        }
-    }
-
-    const addToFollowers = async () => {
-        try {
-            const userRef = doc(db, `users/${id}`);
-            const res = await updateDoc(userRef, {
-                followers: arrayUnion(user.uid)
-            })
-        } catch (e) {
-            console.log(e)
-        }
-    }
-
-    const removeFromFollowing = async () => {
-        try {
-            const userRef = doc(db, `users/${user.id}`);
-            const res = await updateDoc(userRef, {
-                following: arrayRemove(uid)
-            });
-            const filteredFollowing = user.following.filter((userID) => userID !== uid);
-            setUser({ ...user, following: filteredFollowing });
-            localStorage.setItem("user", JSON.stringify({ ...user, following: filteredFollowing }))
-        } catch (e) {
-            console.log(e)
-        }
-    }
-
-    const removeFromFollowers = async () => {
-        try {
-            const userRef = doc(db, `users/${id}`);
-            const res = await updateDoc(userRef, {
-                followers: arrayRemove(user.uid)
-            });
-        } catch (e) {
-            console.log(e)
-        }
-    }
+    const user = useSelector((state) => state.user);
```

Comment:
Instead of repeating this logic that you pass within useSelector parameter in your user Slice file you can write this logic of returning 
```
//In the user Slice file
const getUser=state=>state.user

//Components where you are trying to access the state:
const userDetails=useSelector(getUser);
```

Cannot remember the exact reference where I studied this if this logic doesn't work out then you can continue using this one will not make much difference